### ISSUE TITLE
Translator's "identifier case" option is confusing

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9760]
With _Identifier Case_ option set to _Unmodified_ I would have expected a statement like:
```sql
create table t(x int)
```
to be "translated" to:
```sql
create table t(x int null)
```
with PostgreSQL as the output dialect. But in fact the two identifiers are rendered in upper case:
```sql
create table T(X int null)
```

It seems like there should probably be an option on how to treat the identifier case with respect to the input dialect. And the identifier case options should by default probably also be "dialect-sensitive"; i.e. be set to something that makes sense for the given input / output dialect.

**Comment by filonik** [comment]
Agreed, setting "Identifier Case" to "Unmodified" should preserve the exact case from the input.

**Comment by lukaseder** [comment]
Looking into this now. The desired behaviour can be achieved by specifying the input dialect, e.g. to PostgreSQL:

<img width="1029" alt="image" src="https://user-images.githubusercontent.com/734593/220866457-ededa821-9620-4a3c-aaf6-d65556fca8f7.png">

That's because the translator uses `Settings.parseNameCase == ParseNameCase.DEFAULT`. This converts unquoted identifiers to the default case of the input dialect, which by default is upper case.

This is not at all intuitive, of course, and should be changed to `ParseNameCase.AS_IS`
